subsys/bluetooth/host/audio/chan.c

```diff
@@ -1071,51 +1067,92 @@ struct bt_conn_iso *bt_audio_chan_bind(struct bt_audio_chan *chan,
 		}
 	}
 
-	conns[0] = chan->conn;
-	chans[0] = chan->iso;
-
-	/* Attempt to bind if not bound yet */
-	if (!chan->iso->conn) {
+	if (!chan->iso->iso) {
+		struct bt_iso_cig_create_param param;
+		struct bt_iso_cig **free_cig;
 		int err;
 
-		err = bt_iso_chan_bind(conns, 1, chans);
-		if (err) {
-			BT_ERR("bt_iso_chan_bind: %d", err);
+		free_cig = NULL;
+		for (int i = 0; i < ARRAY_SIZE(cigs); i++) {
+			if (cigs[i] == NULL) {
+				free_cig = &cigs[i];
+				break;
+			}
+		}
+
+		if (free_cig == NULL) {
+			return NULL;
+		}
+
+		/* TODO: Support more than a single CIS in a CIG */
+		param.num_cis = 1;
+		param.cis_channels = &chan->iso;
+		param.framing = qos->framing;
+		param.packing = 0; /*  TODO: Add to QoS struct */
+		param.interval = qos->interval;
+		param.latency = qos->latency;
+		param.sca = BT_GAP_SCA_UNKNOWN;
+
+		err = bt_iso_cig_create(&param, free_cig);
+		if (err != 0) {
+			BT_ERR("bt_iso_cig_create failed: %d", err);
 			return NULL;
 		}
 	}
 
-	return &chan->iso->conn->iso;
+	return &chan->iso->iso->iso;
```

Comment:
I still find these triplicates confusing ...

Reply:
Agreed. 

I'd rather not change the chan->iso in this PR as this is just fixing stuff. 

The other `iso->iso` could probably also be improved upon in some way, but what is upstream.